presto-expressions/src/main/java/com/facebook/presto/expressions/DynamicFilters.java

```diff
@@ -106,20 +115,41 @@ public static boolean isDynamicFilter(RowExpression expression)
         }
 
         CallExpression call = (CallExpression) expression;
+        List<RowExpression> arguments = call.getArguments();
 
         if (!call.getDisplayName().equals(DynamicFilterPlaceholderFunction.NAME)) {
             return Optional.empty();
         }
 
-        List<RowExpression> arguments = call.getArguments();
-        checkArgument(arguments.size() == 2, "invalid arguments count: %s", arguments.size());
+        checkArgument(arguments.size() == 3, "invalid arguments count: %s", arguments.size());
 
-        RowExpression firstArgument = arguments.get(0);
-        checkArgument(firstArgument instanceof ConstantExpression);
-        checkArgument(firstArgument.getType() instanceof VarcharType);
+        RowExpression probeSymbol = arguments.get(0);
+        RowExpression operatorExpression = arguments.get(1);
+        checkArgument(operatorExpression instanceof ConstantExpression);
+        checkArgument(operatorExpression.getType() instanceof VarcharType);
+        String operator = ((Slice) ((ConstantExpression) operatorExpression).getValue()).toStringUtf8();
 
-        String id = ((Slice) ((ConstantExpression) firstArgument).getValue()).toStringUtf8();
-        return Optional.of(new DynamicFilterPlaceholder(id, arguments.get(1)));
+        RowExpression idExpression = arguments.get(2);
+        checkArgument(idExpression instanceof ConstantExpression);
+        checkArgument(idExpression.getType() instanceof VarcharType);
+        String id = ((Slice) ((ConstantExpression) idExpression).getValue()).toStringUtf8();
+
+        if (operator.equals(EQUAL.name())) {
```

Comment:
Nits: maybe put this into a static map so the code can be reduced to a single lookup.